src/os_posix/os_dir.c

```diff
@@ -64,22 +84,37 @@ __directory_list_worker(WT_FILE_SYSTEM *file_system, WT_SESSION *wt_session, con
         if (single)
             break;
     }
-
+    /*
+     * Reading the directory returns NULL on failure or reaching the end of the list. So set the
+     * value into ret as soon as the loop exits.
+     */
+    ret = errno;
     *dirlistp = entries;
     *countp = count;
 
 err:
+    __wt_epoch(session, &ts);
+    WT_TRET(__wt_snprintf(closemsg, sizeof(closemsg),
+      "[%" PRIuMAX ":%" PRIuMAX "] closedir (%s) ret %d dir fd %d", (uintmax_t)ts.tv_sec,
+      (uintmax_t)ts.tv_nsec / WT_THOUSAND, directory, ret, dirfd(dirp)));
     WT_SYSCALL(closedir(dirp), tret);
     if (tret != 0) {
         __wt_err(session, tret, "%s: directory-list: closedir", directory);
         if (ret == 0)
             ret = tret;
+        /* If we have an error print information about the run. */
+        if (open_ready)
+            __wt_errx(session, "%s", openmsg);
+        if (read_ready)
+            __wt_errx(session, "%s", readmsg);
+        __wt_errx(session, "%s", closemsg);
     }
 
     if (ret == 0)
         return (0);
 
-    WT_TRET(__wti_posix_directory_list_free(file_system, wt_session, entries, count));
+    if (open_ready)
```

Comment:
This doesn't need to be changed to be conditional on open_ready.  The free will work if entries is NULL.

Reply:
Fixed.